force-app/main/default/lwc/donationHistoryTable/donationHistoryTable.js

```diff
@@ -65,17 +70,19 @@ export default class DonationHistoryTable extends LightningElement {
      */
     loadMoreDonationData(event){
         event.target.isLoading = true;
-        if (this.data.length >= this.totalNumberOfRows) {
-            event.target.enableInfiniteLoading = false;
-            event.target.isLoading = false;
-            return;
-        }
-        const current = this.data.length;
-        const offset = current + RECORDS_TO_LOAD;
-        const currentData = this.data;
-        //Appends new data to the end of the table
-        const newData = currentData.concat(this.allData.slice(current, offset));
-        this.data = newData;
-        event.target.isLoading = false;
+        this.tableElement = event.target;
+        getDonationHistory({contactId: this.contactId, offset: this.data.length})
+        .then(data => {
+                if (this.data.length >= this.totalNumberOfRows) {
+                    this.tableElement.isLoading = false;
+                    this.tableElement.enableInfiniteLoading = false;
```

Comment:
I see that the docs for lightning-datatable suggest using `event.target`, but I think it's better to bind properties in the template like this:

```suggestion
                    this.isLoading = false;
                    this.enableInfiniteLoading = false;
```

Then in the template add: `enable-infinite-loading=(enableInfiniteLoading}`

Reply:
when implementing this the component does not take the assigned values and apparently the reference is not working, both for isLoading and for enableInfiniteLoading